### ISSUE TITLE
Clippy broken on nightly? (linux-x64)

### ISSUE THREAD

**Issue by stevedonovan** [Manishearth/rust-clippy#1343]
Fresh off the press nightly, and no happiness yet!

```
error[E0053]: method `check_fn` has an incompatible type for trait
  --> /home/steve/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.98/src/unused_label.rs:50:5
   |
50 |     fn check_fn(&mut self, cx: &LateContext, kind: FnKind, decl: &hir::FnDecl, body: &hir::Block, span: Span, fn_id: ast::NodeId) {
   |     ^ expected struct `rustc::hir::Expr`, found struct `rustc::hir::Block`
   |
   = note: expected type `fn(&mut unused_label::UnusedLabel, &rustc::lint::LateContext<'_, '_>, rustc::hir::intravisit::FnKind<'_>, &rustc::hir::FnDecl, &rustc::hir::Expr, syntax::codemap::Span, syntax::ast::NodeId)`
   = note:    found type `fn(&mut unused_label::UnusedLabel, &rustc::lint::LateContext<'_, '_>, rustc::hir::intravisit::FnKind<'_>, &rustc::hir::FnDecl, &rustc::hir::Block, syntax::codemap::Span, syntax::ast::NodeId)`

```

**Comment by oli-obk** [comment]
Sorry, am already on it

**Comment by mcarton** [comment]
Closed by #1344.
